werkzeug/_reloader.py

```diff
@@ -197,10 +204,22 @@ def run(self):
                     try:
                         watches[path] = observer.schedule(
                             self.event_handler, path, recursive=True)
-                    except OSError:
-                        # "Path is not a directory". We could filter out
-                        # those paths beforehand, but that would cause
-                        # additional stat calls.
+                    except OSError as e:
+                        # Extract message from exception
+                        message = str(e)
+
+                        # TODO: This check is hardcoded, should find
+                        # better way to differentiate errors from watchdog
+                        if message == "Path is not a directory":
+                            # "Path is not a directory". We could filter out
+                            # those paths beforehand, but that would cause
+                            # additional stat calls.
+                            message = '%s is not a directory' % path
```

Comment:
To clarify, the behavior I want:

* Ignore "path is not a directory"
* Do `_log('error', messsage)` for everything else.